### PR TITLE
Add hooks for plugins to get control in transientrel_init/intorel_initplan.

### PR DESCRIPTION
**Pull request by HuSen8891**
closes: #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to @cloudberrydb/dev team for review and approval when your PR is ready🥳

### CODE REVIEW

File: src/backend/commands/createas.c [hunk 1]
```diff
@@ -611,11 +592,6 @@ intorel_initplan(struct QueryDesc *queryDesc, int eflags)
 	int			attnum;
 	TupleDesc   typeinfo = queryDesc->tupDesc;
 
-	/* If EXPLAIN/QE, skip creating the "into" relation. */
-	if ((eflags & EXEC_FLAG_EXPLAIN_ONLY) ||
```
Comment (my-ship-it) [inline src/backend/commands/createas.c:615]: Why we move these checks out?

File: src/backend/commands/matview.c [hunk 2]
```diff
@@ -602,6 +588,15 @@ transientrel_init(QueryDesc *queryDesc)
 	heap_close(matviewRel, NoLock);
 }
 
+void
+transientrel_init(QueryDesc *queryDesc)
+{
+	if (transientrel_init_hook)
+		return (*transientrel_init_hook)(queryDesc);
+
+	transientrel_init_internal(queryDesc);
```
Comment (my-ship-it) [inline src/backend/commands/matview.c:597]: Should we create a new file based DestReceiver for QE so that we don't need to hack transientrel_init here?

**Review by my-ship-it — APPROVED** [review]

File: src/backend/commands/matview.c (hunk 2 above)
Comment (HuSen8891) [reply src/backend/commands/matview.c:597]: We can add new DestReceiver, but we also need to hook cloudberry to use this new DestReceiver.

File: src/backend/commands/createas.c (hunk 1 above)
Comment (HuSen8891) [reply src/backend/commands/createas.c:615]: make sanity check first, we can skip stepping into the function for some case.

**Pull request merged**